src/components/CopyToClipboard/copyText.ts

```diff
@@ -1,11 +1,43 @@
 export function copyText(text: string) {
-    if (navigator?.clipboard?.writeText) {
+    if (typeof navigator !== 'undefined' && navigator.clipboard?.writeText) {
         try {
             return navigator.clipboard.writeText(text);
         } catch (error) {
-            return Promise.reject(error);
+            return typeof document === 'undefined' ? Promise.reject(error) : copyTextFallback(text);
         }
     }
 
-    return Promise.reject(new Error('Native copy is not available'));
+    if (typeof document !== 'undefined') {
+        return copyTextFallback(text);
+    }
+
+    return Promise.reject(new Error('Neither navigator.clipboard nor document is available'));
+}
+
+async function copyTextFallback(text: string) {
+    const activeElement = document.activeElement as HTMLElement;
```

Comment:
may be check for instance instead of casting?

```
const activeElement = document.activeElement;

// ...

if (activeElement instanceof HTMLElement) {
  activeElement.focus();
}
```